### DISCUSSION TITLE
Sofar Solar HYD 15KTL-3PH with communication serial 21xxxxxx

### DISCUSSION THREAD

**Discussion by ruzickav** [StephanJoubert/home_assistant_solarman#368] [General]
Hi all,
i have 4 HYD 15KTL-3PH with communication serial 21xxxxxx. I have tryied everything but it is not working winth Home assistant 2023.7.3.

I have tryied to cut number of sensors to just a few - still no connection.
In a log there is an error "Querying registers [1412 - 1417] failed, aborting."

So - I think the problem is in a protocol itselves. There is no need for serial number if we are talking about Modbus TCP. So maybe this is the issue somehow?
Modbus on my inverters are working. I am able to read values with "Modbus Client X" program without difficulties. no needs for Swerial Number.

And I have found the Modbus registers specification for my inverter on this page in xls:
https://github.com/Nedel124/Sofar_G3_LSW3 
Those registers are looks like working and in a good order. I am not able to create new parameters.yaml from this file, but someone may be...

**Comment by Hetzi74** [comment]
Maybe take a look at my setup and solution for my Azzurro branded inverter. 

https://github.com/StephanJoubert/home_assistant_solarman/discussions/371

But I'm not sure if my inverter is really the rebranded version of your inverter, or if it's a different type again.
